cli/src/complete.rs

```diff
@@ -407,14 +416,31 @@ fn all_files_from_rev(rev: String) -> Vec<CompletionCandidate> {
             .map_err(user_error)?;
         let stdout = String::from_utf8_lossy(&output.stdout);
 
-        Ok(stdout.lines().map(CompletionCandidate::new).collect())
+        Ok(stdout
+            .lines()
+            .filter_map(|path| {
+                if !path.starts_with(current) {
```

Comment:
Instead of filtering afterwards here, we should pass the current path as an argument to `jj file list`. That way we can avoid walking parts of the tree that doesn't match. We also need to give `jj file list` (or `jj debug tree`?) an option to also list directory entries so we don't have to list all files in subdirectories. But that can be fixed later.

Reply:
There is a small problem with passing `current` as an argument to `jj file list`. It only works properly if `current` is a correct directory path, not a partial one. For example, in the jj repo:

- `jj file list cli` returns all files in `cli/` as expected
- `jj file list cl` doesn't do that. It doesn't return any files, because there is no directory called `cl/`.

Do you think I should implement some logic to figure out if `current` is a correct directory path and can be passed to `jj file list` as-is? Or if it needs to be truncated to the next parent or something?

Reply:
I think you can turn the path into glob pattern, something like `glob:{escape(path)}*/**`.